### PR TITLE
Chat-monitor: Centralize functions and migrate to JSON

### PR DESCRIPTION
**Pull request by mia-pi-git**
I centralized the code to make it easier to use, but the main purpose of this PR is to move to safer loading code (see comment at 130) and to a safer loading format (JSON) to prevent current issues. More details on the issue in #private-dev.

### CODE REVIEW

File: server/chat-plugins/chat-monitor.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import {FS, Utils} from '../../lib';
 import type {FilterWord} from '../chat';
 
-const MONITOR_FILE = 'config/chat-plugins/chat-monitor.tsv';
+const LEGACY_MONITOR_FILE = 'config/chat-plugins/chat-monitor.tsv';
+const MONITOR_FILE = 'config/chat-plugins/chat-filter.tsv';
```
Comment (KrisXV) [inline server/chat-plugins/chat-monitor.ts:5]:
```suggestion
const MONITOR_FILE = 'config/chat-plugins/chat-filter.json';
```
?

Comment (mia-pi-git) [reply server/chat-plugins/chat-monitor.ts:5]: whoops, typo

File: server/chat-plugins/chat-monitor.ts [hunk 2]
```diff
@@ -45,74 +46,155 @@ const EVASION_DETECTION_SUBSTITUTIONS: {[k: string]: string[]} = {
...
-export function generateRegex(word: string, isEvasion = false, isShortener = false, isReplacement = false) {
-	try {
-		if (isEvasion) {
-			return constructEvasionRegex(word);
-		} else {
-			return new RegExp((isShortener ? `\\b${word}` : word), (isReplacement ? 'igu' : 'iu'));
+		const data = JSON.parse(FS(MONITOR_FILE).readIfExistsSync() || "{}");
+		for (const k in data) {
+			filterWords[k] = [];
+			// previously, this checked to be sure the monitor existed in Chat.monitors and that there was
```
Comment (DaWoblefet) [inline server/chat-plugins/chat-monitor.ts:130]: /* */ yn

Comment (Zarel) [reply server/chat-plugins/chat-monitor.ts:130]: I think a row of `//`s is fine; I don't think we have a use case for `/*` other than doc comments where they're `/**`.

**Comment by Zarel** [comment]
Sure, let's try this out.

**Pull request merged**